app/controllers/media/objects_controller.rb

```diff
@@ -5,9 +5,15 @@ class Media::ObjectsController < ApplicationController
   def show
     style = params[:style]
     @answer = @media_object.answer
-    @response = @answer.response
+    @response = @answer.try(:response)
     disposition = (params[:dl] == "1") ? "attachment" : "inline"
-    authorize! :show, @response
+
+    if @response
+      authorize! :show, @response
+    else
+      authorized = @media_object.token == params[:token]
```

Comment:
Why is it necessary to assign this to a var? Could just be an `unless` directly?